gr-utils/python/blocktool/cli/cli.py

```diff
@@ -0,0 +1,138 @@
+@click.option('-c', '--blocktool-comments', is_flag=True,
+              help='blocktool helper comments will be added in the header file')
+@click.option('-o', '--output', is_flag=True,
+              help='If given, a file with desired output format will be generated')
+def cli(**kwargs):
+    """
+    Block header parsing tool.
+    """
+    kwargs['modtool'] = False
+    if os.path.isfile(kwargs['file_path']):
+        parser = BlockHeaderParser(**kwargs)
+        parser.cli = True
+        run_blocktool(parser)
+        if kwargs['yaml_confirm']:
+            parser.yaml_confirm = True
+            yaml_generator(parser, **kwargs)
+        else:
+            parser.json_confirm = True
+            json_generator(parser, **kwargs)
+    elif os.path.isdir(kwargs['file_path']):
+        parse_directory(**kwargs)
+    else:
+        raise BlockToolException('Invalid file or directory path.')
+
+
+def json_generator(parser, **kwargs):
+    """
+    Generate JSON file for the block header
+    """
+    header = parser.filename.split('.')[0]
+    block = parser.modname.split('-')[-1]
+    if kwargs['output']:
+        json_file = os.path.join('.', block+'_'+header + '.json')
+        with open(json_file, 'w') as _file:
+            json.dump(parser.parsed_data, _file, indent=4)
+    else:
+        print(json.dumps(parser.parsed_data, indent=4))
+
+
+def parse_directory(**kwargs):
+    """
+    Get parsed json and yaml output for complete header directory
+    """
+    kwargs['output'] = True
+    dir_path = kwargs['file_path']
+    dir_path = os.path.abspath(dir_path)
+    list_header = []
+    dir_name = os.path.basename(dir_path)
+    for _header in os.listdir(dir_path):
+        if _header.endswith('.h') and os.path.isfile(os.path.join(dir_path, _header)):
+            list_header.append(os.path.join(dir_path, _header))
+    list_header = sorted(list_header)
+    if list_header:
+        for header_path in list_header:
+            kwargs['file_path'] = header_path
+            header = os.path.basename(header_path)
+            try:
+                parse_dir = BlockHeaderParser(**kwargs)
+                parse_dir.yaml_confirm = True
+                parse_dir.json_confirm = True
+                run_blocktool(parse_dir)
+                yaml_generator(parse_dir, **kwargs)
+                if not kwargs['modtool']:
+                    json_generator(parse_dir, **kwargs)
+            except:
+                logging.basicConfig(level=logging.DEBUG,
+                                    filename=os.path.join('.', dir_name+'_log.out'))
+                logging.exception(
+                    'Log for Exception raised for the header: {}\n'.format(header))
+                click.secho('Parsing unsuccessful: {}'.format(
+                    header), fg='yellow')
+    else:
+        raise BlockToolException(
+            'Invalid directory! No header found to be parsed')
```

Comment:
since there is only one module left in the cli subpackage, lets make it a single module and move cli/cli.py to cli.py